### DISCUSSION TITLE
Choose a new workflow Correlation ID

### DISCUSSION THREAD

**Discussion by fulganta** [elsa-workflows/elsa-core#3801] [General]
Hi,

Is it possible when triggering a new workflow to choose which correlation id it should be using?
Our goal is to have multiple workflows with the same CORRELATION ID in order to follow them better as they are all link to the same transaction, but just different actions on it. We don't want child/parent workflow here. 

Thx

Cyil

**Comment by sfmskywalker** [comment]
For Elsa 2, I don't think this is properly supported (although I think you should try and see what happens). I think there is some logic, somewhere, that determines if an existing workflow instance already exists for a given correlation ID, without distinguishing between workflow definition IDs. I could be wrong though, so I would just try it.

In Elsa 3, having multiple workflows sharing the same correlation ID is definitely supported.
